### PR TITLE
PHP-1370: Implement get/setReadPreference() for MongoCommandCursor

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHP-1370

### CODE REVIEW

File: cursor.c [hunk 1]
```diff
@@ -1232,11 +1202,11 @@ static zend_function_entry MongoCursor_methods[] = {
...
 
-	/* read preferences */
-	PHP_ME(MongoCursor, getReadPreference, arginfo_no_parameters, ZEND_ACC_PUBLIC)
-	PHP_ME(MongoCursor, setReadPreference, arginfo_setReadPreference, ZEND_ACC_PUBLIC)
+	/* read preferences, code is shared through MongoCursorInterface */
+	PHP_ME(MongoCursorInterface, getReadPreference, arginfo_no_parameters, ZEND_ACC_PUBLIC)
+	PHP_ME(MongoCursorInterface, setReadPreference, arginfo_setReadPreference, ZEND_ACC_PUBLIC)
 
-	/* query */
+	/* query, code is shared through MongoCursorInterface */
```
Comment (derickr) [inline cursor.c]: actually, doQuery() is not... so this looks odd now.

File: cursor_shared.c [hunk 2]
```diff
@@ -708,6 +717,38 @@ PHP_METHOD(MongoCursorInterface, dead)
...
+{
+	mongo_cursor *cursor = (mongo_cursor*)zend_object_store_get_object(getThis() TSRMLS_CC);
+
+	array_init(return_value);
+	add_assoc_string(return_value, "type", mongo_read_preference_type_to_name(cursor->read_pref.type), 1);
+	php_mongo_add_tagsets(return_value, &cursor->read_pref);
+}
+/* }}} */
+
+/* {{{ proto bool MongoCursor::setReadPreference(string read_preference [, array tags ])
```
Comment (derickr) [inline cursor_shared.c]: protos are wrong now, they miss "Interface". Variable names also should have a $ in front of them (I think)

File: tests/generic/mongocursor-getreadpreference-001.phpt [hunk 3]
```diff
@@ -24,7 +24,7 @@ $tagParams = array(
 foreach ($modes as $mode) {
     foreach ($tagParams as $tagParam) {
         $m = new MongoClient($baseString . $mode . $tagParam, array('connect' => false));
-        $rp = $m->phpunit->test->getReadPreference();
+        $rp = $m->phpunit->test->find()->getReadPreference();
```
Comment (derickr) [inline tests/generic/mongocursor-getreadpreference-001.phpt]: Why this change? Shouldn't the old one work too?

Comment (jmikola) [inline tests/generic/mongocursor-getreadpreference-001.phpt]: Before this change, this was duplicating the MongoCollection test.

File: cursor_shared.c (hunk 2 above)
Comment (jmikola) [inline cursor_shared.c]: Proto should have MongoCursorInterface, but they don't get `$` before variables (based on prior art).

**Comment by derickr** [comment]
LGTM after fixing the protos.
